internal/configprovider/manager_test.go

```diff
@@ -207,6 +207,49 @@ func TestConfigSourceManager_ResolveErrors(t *testing.T) {
 	}
 }
 
+func TestConfigSourceManager_YAMLInjection(t *testing.T) {
+	ctx := context.Background()
+	manager := newManager(map[string]configsource.ConfigSource{
+		"tstcfgsrc": &testConfigSource{
+			ValueMap: map[string]valueEntry{
+				"valid_yaml_str": {Value: `
+bool: true
+int: 42
+source: string
+map:
+  k0: v0
+  k1: v1
+`},
+				"invalid_yaml_str": {Value: ":"},
+				"valid_yaml_byte_slice": {Value: []byte(`
+bool: true
+int: 42
+source: "[]byte"
+map:
+  k0: v0
+  k1: v1
+`)},
+				"invalid_yaml_byte_slice": {Value: []byte(":")},
+			},
+		},
+	})
+
+	file := path.Join("testdata", "yaml_injection.yaml")
+	cp, err := config.NewParserFromFile(file)
+	require.NoError(t, err)
+
+	expectedFile := path.Join("testdata", "yaml_injection_expected.yaml")
+	expectedParser, err := config.NewParserFromFile(expectedFile)
+	require.NoError(t, err)
+	expectedCfg := expectedParser.Viper().AllSettings()
```

Comment:
If I'm following, the expected `AllSettings()` doesn't seem to be an item under test, so would it make more sense to define it here as a map so as not to allow potential deviations from the expected format?  If `expectedParser.Viper().AllSettings()` is actually being tested as well, then this makes sense as is.

Reply:
What needs to be tested in this condition is that the actual and expected parsed via `config.Parser` are a match.

Reply:
This was also being used in other tests taking the chance to remove it.